internal/config/validation.go

```diff
@@ -96,6 +97,53 @@ func DiscardNativeOnly(c ClusterResources) error {
 	return nil
 }
 
+// validateConfig is meant to validate all the inter-dependencies of a parsed configuration.
+// In this case, we ensure that bfd echo is not enabled on a pool.
+func validateConfig(cfg *Config) error {
+outer:
+	for _, p := range cfg.Pools.ByName {
+		containsV6 := false
+		for _, cidr := range p.CIDR {
+			if ipfamily.ForCIDR(cidr) == ipfamily.IPv6 {
+				containsV6 = true
+				break
+			}
+			if !containsV6 { // we only care about v6 advertisements
+				continue outer
+			}
+		}
+		for _, a := range p.BGPAdvertisements {
+			if len(a.Peers) == 0 {
+				for _, peer := range cfg.Peers {
+					if hasBFDEcho(peer, cfg.BFDProfiles) {
+						return fmt.Errorf("pool %s has bgpadvertisement %s which references peer %s which has bfd echo enabled, which is not possible", p.Name, a.Name, peer.Name)
+					}
+				}
+				continue
+			}
+			for _, peerName := range a.Peers {
+				if peer, ok := cfg.Peers[peerName]; ok {
+					if hasBFDEcho(peer, cfg.BFDProfiles) {
+						return fmt.Errorf("pool %s has bgpadvertisement %s which references peer %s which has bfd echo enabled, which is not possible", p.Name, a.Name, peer.Name)
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func hasBFDEcho(peer *Peer, bfdProfiles map[string]*BFDProfile) bool {
+	profile, ok := bfdProfiles[peer.BFDProfile]
+	if !ok {
+		return false
+	}
+	if profile.EchoMode {
+		return true
+	}
+	return false
```

Comment:
super nit: return profile.EchoMode instead?

Reply:
not sure I like it more, but I can change it